pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGeneratorFunctors.hpp

```diff
@@ -461,4 +608,40 @@ PrecisionT applyNamedGenerator(const GeneratorOperation generator_op,
     }
 }
 
+template <class ExecutionSpace, class PrecisionT>
+PrecisionT applyNCNamedGenerator(
+    const ControlledGeneratorOperation generator_op,
+    Kokkos::View<Kokkos::complex<PrecisionT> *> arr_, std::size_t num_qubits,
+    const std::vector<std::size_t> &controlled_wires,
+    const std::vector<bool> &controlled_values,
+    const std::vector<std::size_t> &wires, bool inverse = false) {
+    switch (generator_op) {
+    case ControlledGeneratorOperation::RX:
+        applyNCGenRX<ExecutionSpace>(arr_, num_qubits, controlled_wires,
+                                     controlled_values, wires, inverse);
+        return -static_cast<PrecisionT>(0.5);
+    case ControlledGeneratorOperation::RY:
+        applyNCGenRY<ExecutionSpace>(arr_, num_qubits, controlled_wires,
+                                     controlled_values, wires, inverse);
+        return -static_cast<PrecisionT>(0.5);
+    case ControlledGeneratorOperation::RZ:
+        applyNCGenRZ<ExecutionSpace>(arr_, num_qubits, controlled_wires,
+                                     controlled_values, wires, inverse);
+        return -static_cast<PrecisionT>(0.5);
+    case ControlledGeneratorOperation::PhaseShift:
+        applyNCGenPhaseShift<ExecutionSpace>(arr_, num_qubits, controlled_wires,
+                                             controlled_values, wires, inverse);
+        return static_cast<PrecisionT>(1.0);
+    case ControlledGeneratorOperation::GlobalPhase:
+        applyNCGenGlobalPhase<ExecutionSpace>(
+            arr_, num_qubits, controlled_wires, controlled_values, wires,
+            inverse);
+        return static_cast<PrecisionT>(-1.0);
+    /// LCOV_EXCL_START
+    default:
+        PL_ABORT("Controlled generator operation does not exist.");
```

Comment:
I suppose we are not testing that. Is it very difficult to do that?

Reply:
I ask because we test the Controlled generator and it is always nice to test all code additions.
Because even if it is a simple piece of code now, it might not be in the future.

Reply:
Actually this is tested here https://github.com/PennyLaneAI/pennylane-lightning/blob/9dd5b448b0d16e2a5e95134dd3233101a597b73f/pennylane_lightning/core/src/simulators/lightning_kokkos/gates/tests/Test_StateVectorKokkos_Generator.cpp#L87 , I don't know why it's in a `LCOV_EXCL`, I will remove it.

Reply:
Nice! Thanks!